src/main/java/org/maxgamer/quickshop/shop/ContainerShop.java

```diff
@@ -1160,35 +1245,93 @@ public DisplayItem getDisplayItem() {
         } else {
             return false;
         }
     }
 
     /**
-     * Returns the shop that shares it's inventory with this one.
-     *
-     * @return the shop that shares it's inventory with this one. Will return null if this shop is not
-     * attached to another.
+     * Updates the attachedShop variable to reflect the currently attached shop, if any.
+     * Also updates the left shop status.
      */
-    public @Nullable ContainerShop getAttachedShop() {
+    public void updateAttachedShop() {
         Util.ensureThread(false);
-        Block c = Util.getSecondHalf(PaperLib.getBlockState(this.getLocation().getBlock(), false).getState());
+        Block c = Util
+            .getSecondHalf(PaperLib.getBlockState(this.getLocation().getBlock(), false).getState());
         if (c == null) {
-            return null;
+            return;
         }
         Shop shop = plugin.getShopManager().getShop(c.getLocation());
-        return shop == null ? null : (ContainerShop) shop;
+        attachedShop = shop == null ? null : (ContainerShop) shop;
+
+        if (attachedShop != null && attachedShop.matches(this.getItem())) {
+            updateLeftShop();
+        } else {
+            attachedShop = null;
+            isLeftShop = false;
+        }
+    }
+
+    /**
+     * This function calculates which block of a double chest is the left block,
+     * relative to the direction the chest is facing. Left shops don't spawn items since
+     * they merge items with the right shop.
+     * It also updates the isLeftShop status of this class to reflect the changes.
+     */
+    private void updateLeftShop() {
+        if (attachedShop == null) {
+            return;
+        }
+
+        switch (((Chest) getLocation().getBlock().getBlockData()).getFacing()) {
+            case WEST:
+                // left block has a smaller z value
+                isLeftShop = getLocation().getZ() < attachedShop.getLocation().getZ();
+                break;
+            case EAST:
+                // left block has a greater z value
+                isLeftShop = getLocation().getZ() > attachedShop.getLocation().getZ();
+                break;
+            case NORTH:
+                // left block has greater x value
+                isLeftShop = getLocation().getX() > attachedShop.getLocation().getX();
+                break;
+            case SOUTH:
+                // left block has a smaller x value
+                isLeftShop = getLocation().getX() < attachedShop.getLocation().getX();
+                break;
+        }
+    }
+
+    /**
+     * Checks to see if it is a real double without updating anything.
+     * @return If the chest is a real double chest, as in it is a double and it has the same item.
+     */
+    public boolean isRealDouble() {
+        Util.ensureThread(false);
+        if (attachedShop == null) {
+            return false;
+        }
+        return attachedShop.matches(this.getItem());
```

Comment:
We should check if the shop was instanceof Chest

Reply:
AttachedShop is always a ContainerShop or null. No need to check.

Reply:
It can be any container, not only for a chest : D

Reply:
Oh I got it, thanks